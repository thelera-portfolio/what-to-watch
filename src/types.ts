interface AuthData {
  login: string,
  password: string,
}

enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH = `NO_AUTH`,
};

interface Comment {
  comment: string,
  date: string,
  id: number,
  rating: number,
  user: {
    id: number,
    name: string,
  },
}

interface Film {
  backgroundColor: string,
  backgroundImage: string,
  description: string,
  director: string,
  genre: string,
  id: number,
  isFavourite: boolean,
  image: string,
  preview: string,
  previewVideoLink: string,
  ratingCount: number,
  ratingScore: number,
  runTime: number,
  starring: string[],
  title: string,
  videoLink: string,
  year: number,
}

enum FilmPageTab {
  OVERVIEW = `Overview`,
  DETAILS = `Details`,
  REVIEWS = `Reviews`,
};

enum Genre {
  ALL = `All genres`,
  ADVENTURE = `Adventure`,
  COMEDIES = `Comedy`,
  CRIME = `Crime`,
  DOCUMENTARY = `Documentary`,
  DRAMAS = `Drama`,
  FANTASY = `Fantasy`,
  HORROR = `Horror`,
  KIDS_AND_FAMILY = `Kid's and Family`,
  ROMANCE = `Romance`,
  SCI_FI = `Sci - Fi`,
  STORY = `Story`,
  THRILLERS = `Thriller`,
};

export {AuthData, AuthorizationStatus, Comment, Film, FilmPageTab, Genre};