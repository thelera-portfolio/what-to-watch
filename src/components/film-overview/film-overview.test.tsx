import * as React from "react";
import * as renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {Film} from "../../types";
import FilmOverview from "./film-overview";

const film: Film = {
  backgroundColor: `#444444`,
  backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  director: `David Yates`,
  genre: `Fantasy`,
  id: 3523525245235,
  isFavourite: true,
  image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  ratingScore: 7.2,
  ratingCount: 248,
  runTime: 223,
  starring: [`Eddie Redmayne`, `Katherine Waterson`, `Dan Folger`],
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  year: 2015,
};

it(`FilmOverview is rendered correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <FilmOverview
          film={film}
        />
      </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
