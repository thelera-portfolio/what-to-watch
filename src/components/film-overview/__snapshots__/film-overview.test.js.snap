// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmOverview is rendered correctly 1`] = `
<div>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      7.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        248
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        David Yates
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Eddie Redmayne, Katherine Waterson, Dan Folger
         and other
      </strong>
    </p>
  </div>
</div>
`;
